woocommerce/woocommerce
# Implement a feature toggle that allows users to switch between the old and new Marketing page

### All Submissions:






### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/33896.

This PR implements a feature toggle at `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features` that allows users to switch between the new and old Marketing page.

For the new `plugins/woocommerce/src/Admin/Features/MultichannelMarketing/Init.php` PHP file (see https://github.com/woocommerce/woocommerce/pull/33964/files#diff-1e22a25c3e56a003f13be4ba230d84e97c355605216e2814100dc83473d33676), most of the code are similar to the existing code that I took inspiration from:

- https://github.com/woocommerce/woocommerce/blob/26df595313761ce264cb7db7a5ad9e2d5a834892/plugins/woocommerce/src/Admin/Features/Navigation/Init.php
- https://github.com/woocommerce/woocommerce/blob/26df595313761ce264cb7db7a5ad9e2d5a834892/plugins/woocommerce/src/Internal/Admin/Analytics.php

📹 Demo video of the feature with my voice:

https://user-images.githubusercontent.com/417342/179581615-8e932fc9-d4f8-4155-8679-f4a2297bce8b.mov

### How to test the changes in this Pull Request:

1. Go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`. You should see the new Marketing feature. It is disabled by default.
2. With the feature disabled, visit the Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. You should see the current existing Marketing page UI.
3. Enable the feature, and visit the Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. You should see the new Marketing page UI, which currently only has a piece of text that says "Multichannel Marketing Placeholder".
4. Disable the feature, and reload the Marketing page again. You should see the current existing Marketing page UI.
5. Repeat the steps above but now with the WooCommerce Navigation experience enabled.

### Other information:




### FOR PR REVIEWER ONLY:
